operators/constellation-node-operator/main.go

```diff
@@ -196,6 +219,11 @@ func main() {
 		os.Exit(1)
 	}
 
+	wg := &sync.WaitGroup{}
```

Comment:
is there a reason to only start the executor here? could it theoretically be possible that NewNodeJoinWatcher triggers the controller before it is started?

Reply:
No particular reason. That's theoretically possible. I'll move it close to `executor.New`